### ISSUE TITLE
OPTIONAL

### ISSUE THREAD

**Issue by Maxhodges** [LiamMartens/sanity-plugin-seo-tools#11]
Hi sorry, I'm having trouble understanding the docs

```// (OPTIONAL) return page description otherwise inferred from scrape```

so I can remove all these optional fields? if I include them, then I need to hard-code their values in my code?

**Comment by LiamMartens** [comment]
Yes you can leave them out, but then it will need to be inferred from the URL scrape, which can take longer and give issues with CORS because it's being accessed from a browser context. These options mostly exist to allow for better checking of the values and get around CORS issues for your website. (feel free to reopen if it is still unclear)

**Comment by Maxhodges** [comment]
thanks, but yeah, I'm still not sure what to do.

shoudl i do somethign like this?

```
   content(doc) {
          return doc.content
        },
        title(doc) {
          return doc.title
          
        },
        description(doc) {
          return doc.description
        },
```
but when I deploy, the seo widget just loads forever and never returns...

**Comment by LiamMartens** [comment]
Yes, the plugin will try to load your `{baseUrl}/{slug}` URL, but this is most likely blocked by CORS because we need to fetch it from the browser. There are 2 ways to solve it
1. Allow CORS from your Sanity studio domain
2. Disable `fetchRemote` altogether and provide all methods manually so the SEO tooling can detect issues in your content straight from the Sanity document (as you detailed yourself)
